src/raw-transaction.ts

```diff
@@ -0,0 +1,43 @@
+import XpringClientErrorMessages from './xpring-client-error-messages'
+import { GetTxResponse } from './generated/web/rpc/v1/tx_pb'
+
+/** Abstraction around raw Transaction Status for compatibility. */
+export interface RawTransactionStatus {
+  getValidated(): boolean
+  getTransactionStatusCode(): string
+  getLastLedgerSequence(): number
+}
+
+/**
+ * A wrapper class which conforms `GetTxResponse` to the `RawTransaction` interface.
+ */
+export default class RawTransaction {
```

Comment:
I'm not in love with this class name. This is an implementation of the RawTransactionStatus class so I think it should be 
<XYZ>RawTransactionStatus.  GetTxResponseRawTransactionStatus sounds like a handful though, which is why I had used the `Wrapper` name before. 

`RawTransaction` sounds like a transaction model object though, when really this is encapsulating a transaction status which is:
- the transaction
- some metadata

Reply:
I agree 100%. I should have tried to make this a separate PR but this is something I really wanted to look into and optimize the functionality of. This is the only class outside of the grpc-network-client that uses the generated buffers which is something that needs to be resolved.